### ISSUE TITLE
Alternate_first_direction bug and potential confusion with rotate_index?

### ISSUE THREAD

**Issue by herrwang0** [mom-ocean/MOM6#1546]
1. There is a bug related to setting first direction from the restart. The following part of the code is within the if(`CS%rotate_index`) block, which means the `First_direction` parameter from the restart is only used to reset G%first_direction when CS%rotate_index is true. 

https://github.com/mom-ocean/MOM6/blob/8d80d64d3387e065d341e1971b28864c71dd97ca/src/core/MOM.F90#L2466-L2469

This could be fixed by moving it outside the block (plus some minor change, depending on the comment below). 

---
2. There might be a bit of confusion with First_direction in restart with `ROTATE_INDEX` turned on. At the moment, when `ROTATE_INDEX` is true, the parameter G%first_direction takes the value of `FIRST_DIRECTION` modified by `INDEX_TURNS`.  
https://github.com/mom-ocean/MOM6/blob/8d80d64d3387e065d341e1971b28864c71dd97ca/src/core/MOM.F90#L2141-L2148

In other words, `FIRST_DIRECTION` specifies the physical first direction, and G%first_direction stores the nominal first direction. Therefore, the question is, which one should we save in the restart?

The former sounds a better choice to me. In that case, an additional parameter needs to be added in subroutine `step_MOM_dynamics`, so that every timestep when CS% first_dir_restart updates, it knows how to translate back to the actual number. (`INDEX_TURNS` is not saved in the control structure).

**Comment by marshallward** [comment]
1. I am guessing that `CS%first_dir_restart` is updated by `MOM_state_initialization()` call ([via ghostly pointer communication set up in `restart_CSp` by `register_restart_field()`](https://github.com/mom-ocean/MOM6/blob/8d80d64d3387e065d341e1971b28864c71dd97ca/src/core/MOM.F90#L3027-L3029).) .  If so, then the restart-based `set_first_direction()` call needs to follow each `MOM_state_initialization()` call.

   There are two `MOM_state_initialization()` calls, `rotate_index` is disabled and enabled, both called under different circumstances.

   So I agree, it looks like the restart flag is not being used for the standard non-rotated case.

    Instead of moving this update outside of the if-block, maybe we should just duplicate the restart-based update to the `CS%rotate_index = .false.` block?  (Or perhaps just move it?  See point 2.)
   
2. I think I agree that the `first_direction` value inside of the `CS%rotate_index = .true.` block probably ought to also be "rotated" by using the `turns` variable (as in L2415).

   But if we're talking about how to to get this working for cases where `rotate_index` is true, then I would suggest we not worry about it for now, since this is primarily a debugging feature, and I cannot imagine a scenario where rotation would be flipped on/off during a restart test with rotation enabled.

   (To tell the truth, I do not think that index rotation even produces the correct restart files at the moment.)

**Comment by herrwang0** [comment]
I agree, and it sounds like it is preferable to treat the `first_direction` from the restart file the same way as the input parameter "FIRST_DIRECTION", in which case we should do a rotate of first direction after it is initialized (when rotate_index = .true.). 

I guess what is slightly more complicated is with `ALTERNATE_FIRST_DIRECTION` turned on. Ideally, we need to keep the value in `CS%first_dir_restart` "unrotated". So if first direction changes every dynamical timestep, we will have to sync `CS%first_dir_restart` and then rotate it back, which means we need to save `turns` in the control structure. 

Alternatively, maybe we could just give a FATAL when `ALTERNATE_FIRST_DIRECTION` and `ROTATE_INDEX` are both on (something similar to the FATAL when PE==1 and ROTATE_INDEX =. true.) Might be a slightly safer solution given that restart with rotation is not quite functional yet?

**Comment by marshallward** [comment]
Raising a `FATAL` when both `ALTERNATE_FIRST_DIRECTION` and `ROTATE_INDEX` are true seems like the better solution to me.  We don't want to lose time or energy working on it.  If it ever becomes important, we can implement it later.

I would just move the [L2466-2469 block](https://github.com/mom-ocean/MOM6/blob/8d80d64d3387e065d341e1971b28864c71dd97ca/src/core/MOM.F90#L2466-L2469
) from the `rotate_index == .true.` to the `.false.` block, and then add your suggested flag test in the `.true.` block.

**Comment by herrwang0** [comment]
Makes sense. I will submit a patch for this. Thanks!
